src/ir/sc.rs

```diff
@@ -11,14 +11,14 @@ use super::{
 
 pub struct SuperCircuit<F, MappingArgs> {
     sub_circuits: Vec<Circuit<F>>,
-    mapping: Option<Box<Mapping<F, MappingArgs>>>,
+    mapping: MappingGenerator<F, MappingArgs>,
 }
```

Comment:
Just curious and for my learning but why are we changing from Box of Mapping to MappingGenerator struct which contains an Rc of Mapping? Is it because mapping now needs multiple ownerships? Why does it need multiple ownerships now?

Reply:
Rc is better when you need to clone because it does not actually clone the element inside. I am not sure if we need the share reference really, but it is harder to find out.

Reply:
I see.